hedgehog/src/Hedgehog/Internal/HTraversable.hs

```diff
@@ -7,5 +7,26 @@ module Hedgehog.Internal.HTraversable (
 
 -- | Higher-order traversable functors.
 --
+-- This is used internally to make symbolic variables concrete given an 'Environment'.
+--
+-- The type checker will ask you to write 'HTraversable' instances for your action types.  If you do
```

Comment:
I think from this line down might be better suited to a tutorial than the documentation.

If you could peel it off for now I would be happy to merge